Source/Core/Core/AchievementManager.cpp

```diff
@@ -131,15 +131,52 @@ void AchievementManager::LoadGameByFilenameAsync(const std::string& iso_path,
 
     const auto fetch_game_data_response = FetchGameData();
     m_is_game_loaded = fetch_game_data_response == ResponseType::SUCCESS;
+
+    // Claim the lock, then queue the fetch unlock data calls, then initialize the unlock map in
+    // ActivateDeactiveAchievements. This allows the calls to process while initializing the
+    // unlock map but then forces them to wait until it's initialized before making modifications to
+    // it.
+    m_lock.lock();
```

Comment:
This can also be a `lock_guard`, just make a scope, ie

```
m_is_game_loaded = fetch_game_data_response == ResponseType::SUCCESS;
{
    std::lock_guard lg{m_lock};
    LoadUnlockData([](ResponseType r_type) {});
    ActivateDeactivateAchievements();
}
callback(fetch_game_data_response);
```